Lib/test/test_queue.py

```diff
@@ -317,38 +317,39 @@ def test_shutdown_all_methods_in_one_thread(self):
     def test_shutdown_immediate_all_methods_in_one_thread(self):
         return self._shutdown_all_methods_in_one_thread(True)
 
-    def _write_msg_thread(self, q, n, results, delay,
-                            i_when_exec_shutdown,
-                            event_start, event_end):
-        event_start.wait()
+    def _write_msg_thread(self, q, n, results,
+                            i_when_exec_shutdown, event_shutdown,
+                            event_start):
+        put_atleast = i_when_exec_shutdown//2
         for i in range(1, n+1):
             try:
                 q.put((i, "YDLO"))
-                results.append(True)
             except self.queue.ShutDown:
                 results.append(False)
-            # triggers shutdown of queue
+                break
+
+            # Be sure that all write_threads
+            # put few items into the queue.
+            if i == put_atleast:
+                event_start.wait()
+
+            # Triggers shutdown of queue.
             if i == i_when_exec_shutdown:
-                event_end.set()
-            time.sleep(delay)
-        # end of all puts
+                if not event_shutdown.is_set():
+                    event_shutdown.set()
+                    results.append(True)
         q.join()
 
-    def _read_msg_thread(self, q, nb, results, delay, event_start):
-        event_start.wait()
-        block = True
-        while nb:
-            time.sleep(delay)
+    def _read_msg_thread(self, q, results, event_start):
+        nbr = 0
```

Comment:
Would be nice if we can make this variable name more descriptive...

Reply:
This variable must be removed. Used for logs to count the number of `get`.